flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -349,6 +349,11 @@ Actor with ActorLogMessages with ActorLogging {
     case Heartbeat(instanceID) =>
       instanceManager.reportHeartBeat(instanceID)
 
+    case RequestStackTrace(instanceID) =>
+      val taskManager = instanceManager.getRegisteredInstanceById(instanceID).getTaskManager
+      val result = AkkaUtils.ask[StackTrace](taskManager, SendStackTrace)
```

Comment:
This is a blocking call within the actor thread. We should avoid this. You can simply forward the ```SendStackTrace``` message to the respective TaskManager: ```taskManager forward SendStacktrace```